Source/Glide64/3dmath.cpp

```diff
@@ -151,48 +151,119 @@ void calc_sphere (VERTEX *v)
 float DotProductC(register float *v1, register float *v2)
 {
     register float result;
-    result = v1[0]*v2[0] + v1[1]*v2[1] + v1[2]*v2[2];
-    return(result);
+
+    result =
+        v1[0] * v2[0]
+      + v1[1] * v2[1]
+      + v1[2] * v2[2]
+    ;
+    return (result);
 }
 
 void NormalizeVectorC(float *v)
 {
     register float len;
-    len = sqrtf(v[0]*v[0] + v[1]*v[1] + v[2]*v[2]);
-    if (len > 0.0f)
-    {
-        v[0] /= len;
-        v[1] /= len;
-        v[2] /= len;
-    }
+
+    len = v[0]*v[0] + v[1]*v[1] + v[2]*v[2];
+    if (len == 0.0f)

```

Comment:
Is this comparison safe? (floating point comparison).
What if you don't get an "exact" zero (say, 0.0000001).